### ISSUE TITLE
array[subscript] is not detected as a use of array

### ISSUE THREAD

**Issue by knirch** [koalaman/shellcheck#102]
Hoping that's the real issue anyway, seen with

``` bash
x=(0 1 2 3)
(( x[1] < 2 ))
```
